clusterloader2/pkg/measurement/common/slos/api_responsiveness_prometheus.go

```diff
@@ -0,0 +1,266 @@
+	latencySamples, err := gatherSamples(c, timeBoundedLatencyQuery)
+	if err != nil {
+		return nil, err
+	}
+	timeBoundedCountQuery := fmt.Sprintf(countQuery, measurementutil.ToPrometheusTime(measuermentDuration))
+	countSamples, err := gatherSamples(c, timeBoundedCountQuery)
+	if err != nil {
+		return nil, err
+	}
+	return a.convertToApiCalls(latencySamples, countSamples)
+}
+
+func (a *apiResponsivenessMeasurementPrometheus) convertToApiCalls(latencySamples, countSamples []*model.Sample) ([]apiCall, error) {
+	apiCalls := make(map[string]*apiCall)
+	ignoredResources := sets.NewString("events")
+	// TODO: figure out why we're getting non-capitalized proxy and fix this.
+	ignoredVerbs := sets.NewString("WATCH", "WATCHLIST", "PROXY", "proxy", "CONNECT")
+
+	for _, sample := range latencySamples {
+		resource := string(sample.Metric["resource"])
+		subresource := string(sample.Metric["subresource"])
+		verb := string(sample.Metric["verb"])
+		scope := string(sample.Metric["scope"])
+		quantile, err := strconv.ParseFloat(string(sample.Metric["quantile"]), 64)
+		if err != nil {
+			return nil, err
+		}
+		if ignoredResources.Has(resource) || ignoredVerbs.Has(verb) {
+			continue
+		}
+
+		latency := time.Duration(float64(sample.Value) * float64(time.Second))
+		addLatency(apiCalls, resource, subresource, verb, scope, quantile, latency)
+	}
+
+	for _, sample := range countSamples {
+		resource := string(sample.Metric["resource"])
+		subresource := string(sample.Metric["subresource"])
+		verb := string(sample.Metric["verb"])
+		scope := string(sample.Metric["scope"])
+		if ignoredResources.Has(resource) || ignoredVerbs.Has(verb) {
+			continue
+		}
+
+		count := int(math.Round(float64(sample.Value)))
+		addCount(apiCalls, resource, subresource, verb, scope, count)
+	}
+
+	var result []apiCall
+	for _, call := range apiCalls {
+		result = append(result, *call)
+	}
+	return result, nil
+}
+
+func addLatency(apiCalls map[string]*apiCall, resource, subresource, verb, scope string, quantile float64, latency time.Duration) {
+	key := getMetricKey(resource, subresource, verb, scope)
+	call, exists := apiCalls[key]
+	if !exists {
+		call = &apiCall{
+			Resource:    resource,
+			Subresource: subresource,
+			Verb:        verb,
+			Scope:       scope,
+		}
+		apiCalls[key] = call
+	}
+	call.Latency.SetQuantile(quantile, latency)
+}
+
+func addCount(apiCalls map[string]*apiCall, resource, subresource, verb, scope string, count int) {
+	if count == 0 {
+		return
+	}
+	key := getMetricKey(resource, subresource, verb, scope)
+	call, exists := apiCalls[key]
+	if !exists {
+		call = &apiCall{
+			Resource:    resource,
+			Subresource: subresource,
+			Verb:        verb,
+			Scope:       scope,
+		}
+		apiCalls[key] = call
+	}
+	call.Count = count
+}
+
+func gatherSamples(c clientset.Interface, query string) ([]*model.Sample, error) {
+	params := map[string]string{"query": query}
+	body, err := c.CoreV1().
```

Comment:
Shouldn't we retry this in case of transient errors, e.g. prometheus temporarily down?

Reply:
`wait.PollImmediate` added.